remove rescues that just re-raise the error

BenMorganIO opened the pull request:
I've been working with with `Spree::Core::Importer::Order` lately and the amount of exception handling has led to some painful debugging. I didn't really understand why the `rescue`s were being called in a large amount of the methods. It was simply capturing a highly detailed ruby message, and then reraising a message that wasn't so helpful. When I submitting a poor payment, the error message was:

``` ruby
2.2.0 :131 > order_importer.create_payments_from_params params, order
RuntimeError: Order import payments: no implicit conversion of Symbol into Integer [:amount, #<BigDecimal:7fdfbc9d54d8,'0.16E2',9(18)>]
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/bundler/gems/spree-47d064d3faa3/core/lib/spree/core/importer/order.rb:199:in `rescue in block in create_payments_from_params'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/bundler/gems/spree-47d064d3faa3/core/lib/spree/core/importer/order.rb:189:in `block in create_payments_from_params'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/bundler/gems/spree-47d064d3faa3/core/lib/spree/core/importer/order.rb:188:in `each'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/bundler/gems/spree-47d064d3faa3/core/lib/spree/core/importer/order.rb:188:in `create_payments_from_params'
    from (irb):131
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/gems/railties-4.2.0/lib/rails/commands/console.rb:110:in `start'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/gems/railties-4.2.0/lib/rails/commands/console.rb:9:in `start'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/gems/zeus-0.15.4/lib/zeus/rails.rb:136:in `console'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/gems/zeus-0.15.4/lib/zeus.rb:148:in `block in command'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/gems/zeus-0.15.4/lib/zeus.rb:135:in `fork'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/gems/zeus-0.15.4/lib/zeus.rb:135:in `command'
    from /Users/benmorgan/.rvm/gems/ruby-2.2.0/gems/zeus-0.15.4/lib/zeus.rb:50:in `go'
    from -e:1:in `<main>'
```

What line did this error occur on? 199 is where the `raise` was and 189 is where the `begin` was. `raise` also adds a lot of overheard in a ruby program. Usually adding a toll to the performance.

This PR removes an anti-patten from Spree.

BenMorganIO committed:
remove rescues that just re-raise the error

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://benmorganio-remove-exception-handling-from-importer.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://benmorganio-remove-exception-handling-from-importer.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

BenMorganIO committed:
use 1.9 syntax

BenMorganIO committed:
add missing space

BenMorganIO commented on core/lib/spree/core/importer/order.rb:
```diff
@@ -52,48 +50,43 @@ def self.create_shipments_from_params(shipments_hash, order)
-              end
+            shipment = order.shipments.build
+            shipment.tracking       = s[:tracking]
+            shipment.stock_location = Spree::StockLocation.find_by_admin_name(s[:stock_location]) || Spree::StockLocation.find_by_name!(s[:stock_location])
```
This line actually seems weird. Why do we find the stock location by find the stock location by finding the stock location name?

JDutil replied:
We try to find it by the `admin_name` and if that doesn't match than we try the `name`. Subtle but different.

BenMorganIO committed:
remove AR rescue

If we aren’t returning the `e` message and there’s nothing actually
nothing to actually `rescue` here except to `raise` a message, then it
will probably add more overhead/debugging time.

BenMorganIO committed:
remove parens to make it within 100 character limit

BenMorganIO committed:
simplify shipment cost update to 79 columns

BenMorganIO committed:
shorten comment to 100 columns

BenMorganIO commented on core/lib/spree/core/importer/order.rb:
```diff
@@ -4,47 +4,45 @@ module Importer
+          end
 
-            params.delete(:user_id) unless user.try(:has_spree_role?, "admin") && params.key?(:user_id)
+          params.delete(:user_id) unless user.try(:has_spree_role?, "admin") && params.key? :user_id
```
oh common!

BenMorganIO committed:
properly remove parens

BenMorganIO commented on core/lib/spree/core/importer/order.rb:
```diff
@@ -52,48 +50,43 @@ def self.create_shipments_from_params(shipments_hash, order)
-            rescue Exception => e
-              raise "Order import shipments: #{e.message} #{s}"
-            end
+            shipping_method = Spree::ShippingMethod.find_by_name(s[:shipping_method]) || Spree::ShippingMethod.find_by_admin_name!(s[:shipping_method])
```
Same goes for here. Why search for the shipping method by searching for the shipping method by searching for the shipping method name?

JDutil replied:
Again looking by `admin_name` or `name`

BenMorganIO commented on core/lib/spree/core/importer/order.rb:
```diff
@@ -169,117 +146,89 @@ def self.create_line_items_from_params(line_items, order)
         end
 
         def self.ensure_country_id_from_params(address)
           return if address.nil? or address[:country_id].present? or address[:country].nil?
```
Does this line actually work? Since we're using the `or` keyword instead of the `||` method, I'm thinking this is causing:

```ruby
(return if address.nil?) || address[:country_id].present? || address[:country].nil?
```

JDutil replied:
Hmm I believe your right. We should definitely be using || as that's what I'd expect it to be, and is listed in our style guide lets switch it.

BenMorganIO committed:
use ensure instead of rescuing * and reraising

BenMorganIO committed:
remove exception handling specs for order importer

BenMorganIO committed:
use rescue instead of ensure

Back to square one…

BenMorganIO committed:
raise the error

gmacdougall commented:
:+1: to removing the unnecessary rescues. I'm curious why you removed the specs which check for the exceptions. I felt like those provided some value.

peterberkenbosch commented:
This had Wombat reasons, that have left my brain currently. Do agree with Gregor about the exception specs. we should still be sure we can handle the exceptions.

BenMorganIO commented:
@gmacdougall they were failing and I thought they were there to ensure the rescue would format the messages. Removed them because I thought I was removing functionality. Do you want me to add them back in?

BenMorganIO commented:
@peterberkenbosch maybe place Wombat specific code somewhere else? We could make a `WombatService` that just calls to the order importer and make the order importer a bit more modularized?

JDutil commented:
I believe the reasoning was that the error messages didn't make sense to people as no one likes reading backtraces.  Would simply removing the begin blocks help provide the better backtrace clarity?  I've gotten lost in this order importer before, and certainly would like to see troubleshooting for it as easy as possible. I don't think overhead from raising the errors is an issue though.

BenMorganIO commented:
I like backtraces...

JDutil commented:
Because you know how to read them ;) Always winds up amazing me when helping people code, and they simply just don't read them.  Granted that was definitely me for awhile thinking umm huge wall of text that's not the "Hello World" I was expecting :)

BenMorganIO committed:
use || instead of or for precedence

BenMorganIO committed:
simplify adjustments and use adjustment instead of a

What is `a`? Is it an anonymous function (lambda)? Is it an albatross?
Oh…. Its an adjustment!

BenMorganIO committed:
rename s to shipment_hash

What is `s`? Is a stock_location? Is it a stock_item? Is it a seal? Is
it a shirt :shirt:? Is it a sum? Is it the country Syria? Nope. Its a
shipment hash!

BenMorganIO committed:
use `line_item` instead of `l`

What is `l`? Is it unconditional love :heart:? Is it a leaf? Is it a
laser? Is it a lion? Nope. Its a line item.

BenMorganIO committed:
missing spots for shipment_hash

BenMorganIO committed:
rename `iu` to `inventory_unit`

What is an `iu`? Is it the International Union? Is it Lee Ji-eun? Is it
Indiana University? Is it the British word “iu”
(http://dictionary.cambridge.org/dictionary/british/iu). Nah. Its an
inventory unit.

BenMorganIO commented:
Can someone remind me on what's needed to get the PR merged?

JDutil commented:
@BenMorganIO I think I need to re-implement wombat error handling or something along those lines for this.  I agree this isn't really the proper place for it, but that is how it got implemented... I'm not sure when I'd be able to get to it either.

BenMorganIO commented:
@jdutil Do you know what the status is with Wombat's Error Handling?
